runtime/near-vm-runner/src/near_vm_runner/metrics.rs

```diff
@@ -0,0 +1,19 @@
+use near_o11y::metrics::{try_create_int_counter_vec, IntCounterVec};
+use once_cell::sync::Lazy;
+
+pub static COMPILED_CONTRACT_CACHE_HIT: Lazy<IntCounterVec> = Lazy::new(|| {
```

Comment:
I think there are [better ways to define metrics in a component like the contract runtime.](https://near.zulipchat.com/#narrow/stream/295306-contract-runtime/topic/Plan.20to.20ensure.20bounded.20apply.20chunk.20latency/near/436044598) Having to add `shard_id` and `apply_reason` to the `VMContext` is a good part of the reason why I don't really love this approach. I don't think VM should need to worry about the reasons why it has been invoked at all -- this sort of the context should be introduced by the caller (somehow.)

I'm working on my change, but I think in the meanwhile it is fine to not block on this specific reason.

Reply:
Updated the PR to use the exported Metrics structure (thread local) to update the hit/miss information. please take another look.